src/storage.rs

```diff
@@ -1,36 +1,49 @@
 use std::{
     fs::{self, DirEntry},
     io,
-    path::Path,
+    path::{Path, PathBuf},
 };
 
 use crate::blob::Blob;
 
+/// # Description
 /// Used to create a storage, configure it and manage
-/// Examples
+/// `T` - type of storage key, must be [Sized](https://doc.rust-lang.org/std/marker/trait.Sized.html)
+/// # Examples
+/// ```
+/// use pearl::{Storage, Builder};
+///
+/// let mut storage = Builder::new().build::<u32>();
+/// storage.init().unwrap();
+/// ```
+///
 #[derive(Debug)]
-pub struct Storage {
+pub struct Storage<T> {
     config: Config,
-    active_blob: Box<Option<Blob>>,
-    blobs: Vec<Blob>,
+    active_blob: Box<Option<Blob<T>>>,
+    blobs: Vec<Blob<T>>,
 }
 
-impl Storage {
-    /// Creates a new instance of a storage
+impl<T> Storage<T>
+where
+    T: Default,
```

Comment:
I think we should define trait for storage key to allow extensibility in the future

Reply:
If you will change Key trait, it'll break backward compatibility. So it is the same as adding new trait if we will need it.